### DISCUSSION TITLE
Zoned::now() with nanoseconds

### DISCUSSION THREAD

**Discussion by fdncred** [BurntSushi/jiff#110] [Q&A] [answered]
I've been trying to get a `Zoned::now()` to output nanoseconds that do not end in 000. I'm wondering if I'm doing something wrong or it's just not possible?

I've tried these two things.
```rust
        let now = Zoned::now();
        let nanos = now.timestamp().as_nanosecond();
        eprintln!("nanos: {:?}", nanos);
```
and
```rust
        let now = Zoned::now().round(Unit::Nanosecond).map_err(|err| {
            LabeledError::new(format!(
                "Error getting current time with Zoned::now(). Error: {}",
                err.to_string()
            ))
        })?;
        let nanos = now.timestamp().as_nanosecond();
        eprintln!("nanos: {:?}", nanos);
```
Example of my output
```nushell
❯ dt now
nanos: 1724106013597000000
Mon, 19 Aug 2024 17:20:13 -0500 (now)
❯ dt now
nanos: 1724106015838647000
Mon, 19 Aug 2024 17:20:15 -0500 (now)
❯ dt now
nanos: 1724106017747553000
Mon, 19 Aug 2024 17:20:17 -0500 (now)
❯ dt now
nanos: 1724106019417424000
Mon, 19 Aug 2024 17:20:19 -0500 (now)
```

**Comment by BurntSushi** [comment, marked as answer]
The most likely explanation is that your platform doesn't have nanosecond resolution. It looks like it only has microsecond resolution. So getting nanoseconds is probably fundamentally impossible as it is a limitation of your hardware.

**Reply by fdncred** [reply]
I was wondering about that. I'm on a M1 Max MacBook Pro MacOS Sonoma 14.6.1. I'll do some web searching to see what Apple has to say. Thanks!
